externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -56,6 +57,16 @@ private[spark] final class SparkSQLEngine(name: String, spark: SparkSession)
     backendService.sessionManager.startTerminatingChecker()
   }
 
+  override def stop(): Unit = {
```

Comment:
If we have EngineServiceDiscovery, do we need to hack here?

Reply:
Seems it's better to move it to `EngineServiceDiscovery`.